### DISCUSSION TITLE
BNN

### DISCUSSION THREAD

**Discussion by DanielMadmon** [tracel-ai/burn#1361] [Q&A] [answered]
Is it possible to implement Binarized Neural Networks in burn?

**Comment by louisfd** [comment, marked as answer]
Yes i think so, I just went through the pseudo code of the paper and it seems really feasible. Maybe you will need to write a backend extension (https://burn.dev/book/advanced/backend-extension/custom-wgpu-kernel.html) for binarize, with your own way of handling the backward pass for this part. We already have batch norm so that should save you time as well.

**Comment by YagaoDirac** [comment]
Hi, @DanielMadmon , I would like to know more about BNN. I believe I tried something similar.
Thanks.
